src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1205,14 +1205,30 @@ def _validate_ssh_key_helper(ssh_key_value, should_generate_ssh_keys):
 
 
 def _validate_vm_vmss_msi(cmd, namespace, from_set_command=False):
+
+    # For the creation of VM and VMSS, "--role" and "--scope" should be passed in at the same time
+    # when assigning a role to the managed identity
+    if not from_set_command and namespace.assign_identity is not None:
+        if namespace.identity_scope and not namespace.identity_role:
+            raise ArgumentUsageError(
+                "usage error: '--scope {}' is not applicable as the '--role' is not provided".format(
+                    namespace.identity_scope))
+
+    # assign managed identity
     if from_set_command or namespace.assign_identity is not None:
         identities = namespace.assign_identity or []
         from ._vm_utils import MSI_LOCAL_ID
         for i, _ in enumerate(identities):
             if identities[i] != MSI_LOCAL_ID:
                 identities[i] = _get_resource_id(cmd.cli_ctx, identities[i], namespace.resource_group_name,
                                                  'userAssignedIdentities', 'Microsoft.ManagedIdentity')
-        if not namespace.identity_scope and getattr(namespace.identity_role, 'is_default', None) is None:
+
+        if from_set_command:
+            role_is_explicitly_specified = getattr(namespace.identity_role, 'is_default', None) is None
+        else:
+            role_is_explicitly_specified = namespace.identity_role is not None
+
+        if not namespace.identity_scope and role_is_explicitly_specified:
             raise ArgumentUsageError("usage error: '--role {}' is not applicable as the '--scope' is not provided".
                                      format(namespace.identity_role))
```

Comment:
Here is the error message that `--scope` parameter is not passed in

Reply:
I mean, I don't like this message either. It is hard to understand.